# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/mansari40/flyrank/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

I am choosing **Lane 2: Refresh / Content Opportunity Scoring**. I want to focus on the practical question of which pages should be reviewed first instead of treating every page as equally important. This lane fits well because it combines search performance, content signals, and machine learning into a ranked review queue. I also like that the result can be explained to a human reviewer through scores, actions, and reason codes rather than just producing a model score.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

The main decision is **which pages should be reviewed first for a possible refresh, improvement, protection, pruning, or monitoring action**. The people using the result would be content or SEO teams who have limited time and need to decide where to focus first. A wrong recommendation can waste review and editing time on a page that does not need attention. On the other hand, ranking a weak page too low could mean missing a useful opportunity. Because of this, I want the ranking to focus on pages with enough evidence behind them and to give a clear reason for why a page was selected.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [12]:
import pandas as pd

df = pd.read_csv("../../data/raw/content_refresh_anonymized.csv")

df = df[
    (df["impressions_90d"] > 0) &
    (df["content_age_days"] >= 90)
].drop_duplicates("content_id")

print(f"Usable pages: {len(df):,}")
print(f"Pages with a downward trend: {(df['trend_direction'] == 'down').mean():.1%}")
print(f"Median 90-day impressions: {df['impressions_90d'].median():,.0f}")
display(df.head())

Usable pages: 30,000
Pages with a downward trend: 54.2%
Median 90-day impressions: 731


,content_id,client_id,search_volume,competition,competition_level,cpc,content_type,main_intent,word_count,char_count,...,char_count_tier,ctr,avg_position,engagement_rate,scroll_rate,ai_traffic_pct,impression_tier,position_tier,trend_direction,trend_pct
0,content_304f48230142,client_f369cb89fc,10.0,0.67,HIGH,2.05,keyword article,transactional,3221.0,20457.0,...,15000-25000,0.76,10.6,5.88,4.55,0.0,good,striking,down,-41.4
1,content_a1fb4e703a9e,client_4e07408562,90.0,0.01,LOW,0.05,keyword article,informational,2481.0,15562.0,...,15000-25000,0.05,20.3,0.00,10.00,0.0,good,page_3_5,down,-57.7
2,content_9aa793d4d895,client_7f2253d7e2,0.0,0.00,LOW,0.00,keyword article,informational,3515.0,23643.0,...,15000-25000,0.09,36.5,0.00,28.57,0.0,good,page_3_5,down,-60.9
3,content_331d6c4de07b,client_19581e27de,10.0,0.00,LOW,0.00,keyword article,commercial,NaN,NaN,...,NaN,0.49,6.2,1.28,3.45,0.0,good,page_1,stable,-13.8
4,content_d99b7a2d90ca,client_3fdba35f04,0.0,0.00,LOW,0.00,keyword article,informational,2803.0,17469.0,...,15000-25000,0.13,44.0,0.00,24.29,0.0,good,page_3_5,down,-34.7


The starter data gives me a good reason to look further into this problem. After applying the starter filters, there are 30,000 usable pages. 54.2% of these pages have a downward trend, based on the current trend label. The median page has 731 impressions over 90 days, so there is enough search activity to compare pages and identify which ones may deserve attention first. These numbers do not prove that a page needs a refresh, but they make the prioritization problem worth investigating.

## 4. Careful words: what I can and can't claim


My work will be able to identify observed patterns and directional relationships in the data and use them to support decisions about which pages should be reviewed first. The model can rank pages and provide reasons for their priority. I cannot claim that any factor causes a change in page performance, that refreshing a page will definitely improve it, or that the model can predict Google's ranking algorithm. The results should be treated as decision support, not causal proof.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.